### PR TITLE
fix: modify build error of ndt related packages in rolling

### PR DESCRIPTION
**Pull request by wep21**
## Description

This PR fixes the build errors of ndt related packages on rolling distro mainly caused by PCL update.
See the links below.
https://github.com/PointCloudLibrary/pcl/commit/88524c1b4b28c64912e1fc619690acb3457ed9fa
https://github.com/PointCloudLibrary/pcl/issues/2792
https://github.com/ros-perception/perception_pcl/pull/361

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/ndt/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(PCL REQUIRED COMPONENTS common io registration)
+find_package(Eigen3 REQUIRED)
```
Comment (wep21) [inline localization/ndt/CMakeLists.txt:33]:
```suggestion
```

File: localization/ndt_pcl_modified/include/ndt_pcl_modified/ndt.hpp [hunk 2]
```diff
@@ -102,7 +102,11 @@ class NormalDistributionsTransformModified
   using NormalDistributionsTransform<PointSource, PointTarget>::step_size_;
   using NormalDistributionsTransform<PointSource, PointTarget>::gauss_d1_;
   using NormalDistributionsTransform<PointSource, PointTarget>::gauss_d2_;
+#if PCL_VERSION >= PCL_VERSION_CALC(1, 12, 1)
+  using NormalDistributionsTransform<PointSource, PointTarget>::point_jacobian_;
```
Comment (kenji-miyake) [inline localization/ndt_pcl_modified/include/ndt_pcl_modified/ndt.hpp:106]: Could you explain this a bit more in detail? :pray:

Comment (wep21) [reply localization/ndt_pcl_modified/include/ndt_pcl_modified/ndt.hpp:106]: You can see the api change in the first link in the description.

Comment (kenji-miyake) [reply localization/ndt_pcl_modified/include/ndt_pcl_modified/ndt.hpp:106]: I see, thank you!
image

**Review by kenji-miyake — APPROVED** [review]

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**